### ISSUE TITLE
CTRL-C causing sessions to freeze on 9.0

### ISSUE THREAD

**Issue by shaneshort** [gravitational/teleport#11092]
### Description

After installing teleport 9.x, when I attempt to send a ctrl-c to the remote terminal the session completely hangs and I have to use the ssh escape sequence ~. to quit and reconnect

**What you expected to happen**:
ctrl-c should be passed through without breaking.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
Simply send ctrl-c to the console

### Server Details
- Teleport version (run `teleport version`):
Teleport v9.0.0 git:v9.0.0-0-g1fa8857aa go1.17.7
- Server OS (e.g. from `/etc/os-release`):
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):
local, inside proxmox
- Additional details:

### Client Details
- Tsh version (`tsh version`):
Teleport v9.0.0 git:v9.0.0-0-g1fa8857aa go1.17.7
- Computer OS (e.g. Linux, macOS, Windows):
Ubuntu 20.04
- Browser version (for UI-related issues):
N/A
- Installed via (e.g. apt, yum, brew, website download):
APT
- Additional details:

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
- Start Teleport with --debug flag (`teleport --debug`)
- Run tsh with --debug flag (`tsh --debug`)

0 log output after connection establishes until such time as I kill the session with the escape sequence.

**Comment by zmb3** [comment]
@xacrimon this seems likely to be related to moderated sessions and the final PR with the terminate notifier. Can you take a look on Monday?

**Comment by xacrimon** [comment]
> @xacrimon this seems likely to be related to moderated sessions and the final PR with the terminate notifier. Can you take a look on Monday?

Will do

**Comment by doyelese** [comment]
Also experiencing this since upgrading

**Comment by xacrimon** [comment]
Yep. Confirmed the issue, a fix should be released in a patch in the coming days.

**Comment by shaneshort** [comment]
excellent, thanks for the prompt turnaround guys!

I don't know if this is related or not, but are the old packages actively removed from the APT repo when a new version is released? It appears to be the case and made rolling back an absolute pain in the ass for us. Any chance we could get the latest stable release of the previous version left behind?

**Comment by slavitch** [comment]
Please leave old packages available for rollback.

**Comment by alexmv** [comment]
Is there an ETA for 9.0.1 with this fix?  This is unfortunately a big usability regression. :(

**Comment by xacrimon** [comment]
We should be cutting 9.0.1 today or tomorrow. @alexmv

**Comment by alexmv** [comment]
Great -- thank you for the quick fix!
